tests/federation_room_join_partial_state_test.go

```diff
@@ -489,6 +490,51 @@ func testReceiveEventDuringPartialStateJoin(
 		t.Fatalf("GET /event failed with %d: %s", res.StatusCode, string(eventResBody))
 	}
 
+	// fire off a /state_ids request for the last event.
+	// it must either:
+	//   * block because the homeserver does not have full state at the last event
+	//   * or 403 because the homeserver does not have full state yet and does not consider the
+	//     Complement homeserver to be in the room
```

Comment:
shouldn't we know which, and test for that specifically?

Reply:
Right now only the last test case blocks instead of 403s. I'm somewhat reluctant to be more specific in the tests, because it feels like an implementation detail of synapse (which will change when we get around to https://github.com/matrix-org/synapse/issues/13288).

If you're really keen on being more specific, I'm happy to add a boolean parameter or suchlike. (It's not easy to move this part of the test out, since it needs to happen after the /event request above has succeeded)

Reply:
ah right. tbh I don't think either blocking or returning a 403 is correct behaviour, though I'm not quite sure what is.

It would be very useful to add a link to #13288 and say that this is somewhat temporary until we solve it properly.

Reply:
hm, is there a better way to test whether synapse thinks an event has partial state?
this part of the test was intended to check that `_resolve_state_at_missing_prevs` calculates the partial state flag correctly

Reply:
none that I can immediately think of. The ideal thing would be to check that *after* the resync, /state_ids returns some result that would have been impossible without the information conveyed during the resync - but that sounds rather fiddly for a test (and, probably, a hard-to-understand test failure if it later turns out not to be true).